sv.c

```diff
@@ -4205,6 +4205,28 @@ S_sv_buf_to_rw(pTHX_ SV *sv)
       && len                    /* and really is a string */               \
     )
 
+/* The test in this macro was also extracted from Perl_sv_setsv_flags so
+ * that it could be used elsewhere. */
+#ifdef PERL_COPY_ON_WRITE
+#define S_SvPV_shared_hkey_or_CoWable(ssv, dsv, sflags, cur, len)          \
+        (sflags & SVf_IsCOW                                                \
+            ? (!len ||                                                     \
+              ( (CHECK_COWBUF_THRESHOLD(cur,len) || SvLEN(dsv) < cur+1)    \
+                  /* If this is a regular (non-hek) COW, only so */        \
+                  /*  many COW "copies" are possible. */                   \
+                && CowREFCNT(ssv) != SV_COW_REFCNT_MAX  ))                 \
+            : ( (sflags & CAN_COW_MASK) == CAN_COW_FLAGS                   \
+                && !(SvFLAGS(dsv) & SVf_BREAK)                             \
+                && CHECK_COW_THRESHOLD(cur,len) && cur+1 < len             \
+                && (CHECK_COWBUF_THRESHOLD(cur,len) || SvLEN(dsv) < cur+1) \
+        ))
```

Comment:
same as above, +1 logic or +2 logic?

Reply:
I'm just extracting this pre-existing condition code. I don't want to change it in this PR. (I suspect it's probably okay as-is, but would have to sit down with lots of post-its to be sure.)

Reply:
If I look around the commit, i see where you copy pasted it from, so I guess its fine for this commit. Any flaws or bugs or concerns about +1 or +2 logic are not part of the PR and are probably already discussed in a random stalled open p5p gh ticket.